### PR TITLE
Chrome Canary Support

### PR DESCRIPTION
**Pull request by ishita12**
## Description
Please describe the change as necessary.
If it's a feature or enhancement please be as detailed as possible.
If it's a bug fix, please link the issue that it fixes or describe the bug in as much detail.

## Specific Changes proposed
Please list the specific changes involved in this pull request.

## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/liwecukasi/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: test/karma.conf.js [hunk 1]
```diff
@@ -8,91 +8,106 @@ module.exports = function(config) {
...
     },
-    babelPreprocessor: {
-      options: {
-        presets: ['es2015'],
-        sourceMap: 'inline'
-      },
-      sourceFileName: function (file) {
-        return file.originalPath;
-      }
+    Chrome_Canary: {
```
Comment (forbesjo) [inline test/karma.conf.js]: Can you rename this to `ChromeCanaryHeadlessNoSandbox `

File: test/karma.conf.js [hunk 2]
```diff
@@ -8,91 +8,106 @@ module.exports = function(config) {
...
-        'test/data/**',
-      ]
+// If no browsers are specified, we enable `karma-detect-browsers`
+// this will detect all browsers that are available for testing
+if (!config.browsers.length) {
+  detectBrowsers.enabled = true;
+}
+
+config.set({
+  basePath: '..',
```
Comment (forbesjo) [inline test/karma.conf.js]: Looks like indentation was shifted here

File: test/karma.conf.js [hunk 3]
```diff
@@ -8,91 +8,106 @@ module.exports = function(config) {
...
       var safariIndex = availableBrowsers.indexOf('Safari');
+      //  var firfoxIndex = availableBrowsers.indexOf('Firefox');
 
+      //var chromeIndex = availableBrowsers.indexOf('Chrome');
       if(safariIndex !== -1) {
         availableBrowsers.splice(safariIndex, 1);
         console.log("Disabled Safari as it was/is not supported");
       }
-      return availableBrowsers;
+      /*  if(firfoxIndex !== -1) {
```
Comment (forbesjo) [inline test/karma.conf.js]: Remove commented code

File: test/karma.conf.js [hunk 4]
```diff
@@ -8,91 +8,106 @@ module.exports = function(config) {
     usePhantomJS: false,
     postDetection: function(availableBrowsers) {
       var safariIndex = availableBrowsers.indexOf('Safari');
+      //  var firfoxIndex = availableBrowsers.indexOf('Firefox');
 
+      //var chromeIndex = availableBrowsers.indexOf('Chrome');
```
Comment (forbesjo) [inline test/karma.conf.js]: Remove commented code

File: .travis.yml [hunk 5]
```diff
@@ -2,4 +2,4 @@ sudo: required
 dist: trusty
 language: node_js
 addons:
-  chrome: stable
+    chrome: stable
```
Comment (forbesjo) [inline .travis.yml]: Fix indentation

File: package-lock.json [hunk 6]
```diff
@@ -4,22 +4,6 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
-    "Base64": {
```
Comment (forbesjo) [inline package-lock.json]: This package-lock file shouldn't need to change

File: test/karma.conf.js [hunk 7]
```diff
@@ -48,7 +34,11 @@ module.exports = function(config) {
       ChromeHeadlessNoSandbox: {
         base: 'ChromeHeadless',
         flags: ['--no-sandbox']
-      }
+      },
+      ChromeCanaryHeadlessNoSandbox: {
+        base: 'ChromeCanaryHeadless',
+        flags: ['--no-sandbox','--autoplay-policy=no-user-gesture-required'],
```
Comment (forbesjo) [inline test/karma.conf.js]: Space between array items

File: test/karma.conf.js [hunk 8]
```diff
@@ -3,33 +3,19 @@ var isparta = require('isparta');
...
-      }
-      return availableBrowsers;
-    }
-  };
 
   if (process.env.TRAVIS) {
     config.browsers = ['ChromeHeadlessNoSandbox'];
   }
+  else {
+    config.browsers=['ChromeCanaryHeadlessNoSandbox','FirefoxHeadless'];
```
Comment (forbesjo) [inline test/karma.conf.js]: Space between equals and array items

File: test/karma.conf.js [hunk 9]
```diff
@@ -3,33 +3,19 @@ var isparta = require('isparta');
...
-        console.log("Disabled Safari as it was/is not supported");
-      }
-      return availableBrowsers;
-    }
-  };
 
   if (process.env.TRAVIS) {
     config.browsers = ['ChromeHeadlessNoSandbox'];
   }
+  else {
```
Comment (forbesjo) [inline test/karma.conf.js]: `} else {`

**Comment by squarebracket** [comment]
Is there something wrong with karma-detect-browsers? I feel like we should be massaging what comes out of detect-browsers instead of hard-coding in a requirement for chrome canary. If karma-chrome-launcher can find `google-chrome-canary` or `google-chrome-unstable`, then karma-detect-browsers will automatically use it.

File: test/karma.conf.js [hunk 10]
```diff
@@ -48,7 +33,11 @@ module.exports = function(config) {
       ChromeHeadlessNoSandbox: {
         base: 'ChromeHeadless',
         flags: ['--no-sandbox']
-      }
+      },
+      ChromeCanaryHeadlessNoSandbox: {
+        base: 'ChromeCanaryHeadless',
+        flags: ['--no-sandbox', '--autoplay-policy=no-user-gesture-required'],
```
[the remark this answers is not in the document]
Comment (ishita12) [reply test/karma.conf.js]: I will change that.

**Pull request merged**